### PR TITLE
Db to mongo

### PR DESCRIPTION
**Pull request by HorGra**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -210,5 +210,10 @@
             <artifactId>commons-math3</artifactId>
             <version>3.5</version>
         </dependency>
+        <dependency>
+            <groupId>org.mongodb</groupId>
+            <artifactId>mongo-java-driver</artifactId>
```
Comment (elbeyati) [inline pom.xml]: Remove all jdbc dependencies.

File: src/main/java/com/kh/lolstats/MongoResetDB.java [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+import com.mongodb.client.MongoCollection;
+import com.mongodb.client.MongoDatabase;
+import org.bson.Document;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+public class MongoResetDB {
+    private static List<String> regionsList = new ArrayList<>(Arrays.asList("BR", "OCE", "NA", "RU", "EUW", "EUNE", "KR", "LAN", "TR", "LAS"));
```
Comment (elbeyati) [inline src/main/java/com/kh/lolstats/MongoResetDB.java]: you dont need to put the type of the data structure in the name of a variable. i.e. these should be called regions and keys.

Comment (elbeyati) [inline src/main/java/com/kh/lolstats/MongoResetDB.java]: There's a Regions enum. Use it.

File: src/main/java/com/kh/lolstats/MongoResetDB.java [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+        MongoClient mongoClient = new MongoClient();
+        MongoDatabase db = mongoClient.getDatabase("ROBOLOL");
+        db.drop();
+        db = mongoClient.getDatabase("ROBOLOL");
+        db.createCollection("ITEMS");
+        db.createCollection("CHAMPIONS");
+        db.createCollection("PATCH_DATES");
+        db.createCollection("MATCHES");
+        db.createCollection("API_KEYS");
+        db.createCollection("REGIONS");
```
Comment (elbeyati) [inline src/main/java/com/kh/lolstats/MongoResetDB.java]: Do we even need a regions collection? Remember mongo isn't relational.

File: src/main/java/com/kh/lolstats/StatsMySqlDaemon.java [hunk 4]
```diff
@@ -174,7 +177,7 @@ private String generateChampStats(ShortSet champIds, ChampQueryService service)
     }
 
     public static void main(String[] args) throws Exception {
-        StatsMySqlDaemon daemon = new StatsMySqlDaemon(args[0], args[1], args[2], Integer.parseInt(args[3]), Integer.parseInt(args[4]), args[5]);
+        StatsMySqlDaemon daemon = new StatsMySqlDaemon(args[0], args[1], args[2], Integer.parseInt(args[3]), Integer.parseInt(args[4]), args[5], args[6]);
```
Comment (elbeyati) [inline src/main/java/com/kh/lolstats/StatsMySqlDaemon.java]: Remove the jdbc args, and put the mongo arg in their position.

File: src/main/java/com/kh/lolstats/Test.java [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+import com.kh.lolstats.parser.DataLoader;
+import com.mongodb.MongoClient;
+import com.mongodb.client.MongoDatabase;
+import net.openhft.koloboke.collect.set.ShortSet;
+import net.openhft.koloboke.function.ShortConsumer;
+
+import java.sql.SQLException;
+
+public class Test {
```
Comment (elbeyati) [inline src/main/java/com/kh/lolstats/Test.java]: Don't commit this.

File: src/main/java/com/kh/lolstats/parser/DataLoader.java [hunk 6]
```diff
@@ -21,113 +21,83 @@
...
-
-                    if (++count % 1000 == 0) {
-                        logger.info("Loading match {}.", count);
-                    }
-
-                    consumer.accept(match);
-                }
-            }
-        }
+//        int count = 1;
```
Comment (elbeyati) [inline src/main/java/com/kh/lolstats/parser/DataLoader.java]: Implement this.

File: src/main/java/com/kh/lolstats/parser/DataExtractor.java [hunk 7]
```diff
@@ -160,7 +160,7 @@ private void dispatchMatch(Match match) {
             return;
         }
 
-        dispatcher.dispatch(match);
+        dispatcher.dispatch(match.getRawJson());
```
Comment (elbeyati) [inline src/main/java/com/kh/lolstats/parser/DataExtractor.java]: The DataExtractor/ApiCaller doesn't need to have knowledge of Match anymore.

File: src/main/java/com/kh/lolstats/persister/Persister.java [hunk 8]
```diff
@@ -62,31 +59,18 @@ private void persist(List<Match> matches) throws InterruptedException {
...
-        }
+    private void persistMatches(List<Document> matches) {
+        matchesCol.insertMany(matches);
     }
 
-    public void dispatch(Match object) {
+    public void dispatch(String rawMatchString) {
         try {
-            queue.put(object);
+            queue.put(Document.parse(rawMatchString));
```
Comment (elbeyati) [inline src/main/java/com/kh/lolstats/persister/Persister.java]: The parsing should be done in the persister Thread. So the queue should still be of type String. We don't want to slow down the data extractor.

**Comment by elbeyati** [comment]
Squash your commits

File: pom.xml (hunk 1 above)
Comment (HorGra) [inline pom.xml]: Removed mysql and h2 dependencies.

File: src/main/java/com/kh/lolstats/MongoResetDB.java (hunk 2 above)
Comment (HorGra) [inline src/main/java/com/kh/lolstats/MongoResetDB.java]: Done

File: src/main/java/com/kh/lolstats/StatsMySqlDaemon.java (hunk 4 above)
Comment (HorGra) [inline src/main/java/com/kh/lolstats/StatsMySqlDaemon.java]: Removed class.

File: src/main/java/com/kh/lolstats/Test.java (hunk 5 above)
Comment (HorGra) [inline src/main/java/com/kh/lolstats/Test.java]: Removed.

File: src/main/java/com/kh/lolstats/parser/DataLoader.java (hunk 6 above)
Comment (HorGra) [inline src/main/java/com/kh/lolstats/parser/DataLoader.java]: Done

File: src/main/java/com/kh/lolstats/persister/Persister.java (hunk 8 above)
Comment (HorGra) [inline src/main/java/com/kh/lolstats/persister/Persister.java]: Moved

**Pull request merged**